library/core/src/iter/traits/iterator.rs

```diff
@@ -727,21 +727,10 @@ pub trait Iterator {
     ///
     /// The closure must return an [`Option<T>`]. `filter_map` creates an
     /// iterator which calls this closure on each element. If the closure
-    /// returns [`Some(element)`][`Some`], then that element is returned. If the
-    /// closure returns [`None`], it will try again, and call the closure on the
-    /// next element, seeing if it will return [`Some`].
-    ///
-    /// Why `filter_map` and not just [`filter`] and [`map`]? The key is in this
-    /// part:
-    ///
-    /// [`filter`]: #method.filter
-    /// [`map`]: #method.map
-    ///
-    /// > If the closure returns [`Some(element)`][`Some`], then that element is returned.
-    ///
-    /// In other words, it removes the [`Option<T>`] layer automatically. If your
-    /// mapping is already returning an [`Option<T>`] and you want to skip over
-    /// [`None`]s, then `filter_map` is much, much nicer to use.
```

Comment:
Where did this section go? This seems useful.

Reply:
I found this part a bit too verbose/conversational, and so condensed it into the main description to make something more concise:

> If the closure returns `Some(element)`, then the mapped element (without the `Some` wrapper) will be present in the returned iterator.

The two paragraphs I removed are captured by `without the Some wrapper` part.

Reply:
But now there's no discussion of _why_ you'd use this method, just what it does. Half the point of documentation is to tell you why you'd use this API.